src/core.jl

```diff
@@ -1230,6 +1230,19 @@ Throws a `NodeNameNotFound` exception if there is no such tensor.
     return Tensor(node, port)
 end
 
+Base.getindex(graph::Graph, name) = get_tensor_by_name(graph, name)
+Base.values(graph::Graph) = get_operations(graph)
+Base.keys(graph::Graph) = (node_name(op) for op in values(graph))
+Base.haskey(graph::Graph, name) = isnull(get_node_by_name(graph, name))
+
+# delegate these operations on the Graph for the session that holds it.
+Base.values(sess::Session) = values(sess.graph)
```

Comment:
I like defining these for `Graph`, but am not sure about `Session` - if we're going to treat `Session`s as dictionaries, maybe the keys should refer to session properties instead of tensors. Maybe let's just comment this out for now.